python_modules/libraries/dagster-k8s/dagster_k8s/job.py

```diff
@@ -446,6 +447,39 @@ def config_type_run_launcher(cls):
                     description="Raw Kubernetes configuration for launched runs.",
                 ),
                 "job_namespace": Field(StringSource, is_required=False, default_value="default"),
+                "only_allow_user_defined_k8s_config_fields": Field(
+                    Shape(
+                        {
+                            "container_config": Field(
+                                Map(key_type=str, inner_type=bool), is_required=False
+                            ),
+                            "pod_spec_config": Field(
+                                Map(key_type=str, inner_type=bool), is_required=False
+                            ),
+                            "pod_template_spec_metadata": Field(
+                                Map(key_type=str, inner_type=bool), is_required=False
+                            ),
+                            "job_metadata": Field(
+                                Map(key_type=str, inner_type=bool), is_required=False
+                            ),
+                            "job_spec_config": Field(
+                                Map(key_type=str, inner_type=bool), is_required=False
+                            ),
+                            "namespace": Field(bool, is_required=False),
```

Comment:
one thing i realized is that namespace can live under job_metadata and pod_template_spec_metadata - it doesn't need to be a top-level field https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1ObjectMeta.md